test/DGmethods/compressible_Navier_Stokes/mms_bc.jl

```diff
@@ -23,116 +23,115 @@ else
   const ArrayTypes = (Array, )
 end
 
-const _nstate = 5
-const _ρ, _U, _V, _W, _E = 1:_nstate
-const stateid = (ρid = _ρ, Uid = _U, Vid = _V, Wid = _W, Eid = _E)
-const statenames = ("ρ", "U", "V", "W", "E")
 
-const _nviscstates = 6
-const _τ11, _τ22, _τ33, _τ12, _τ13, _τ23 = 1:_nviscstates
-
-const _ngradstates = 3
-const _states_for_gradient_transform = (_ρ, _U, _V, _W)
+struct AtmosModel <: BalanceLaw
+end
 
-if !@isdefined integration_testing
-  const integration_testing =
-    parse(Bool, lowercase(get(ENV,"JULIA_CLIMA_INTEGRATION_TESTING","false")))
+dimension(::AtmosModel) = 3
+vars_aux(::AtmosModel) = ()
+vars_state(::AtmosModel) = (:ρ, :ρu, :ρv, :ρw, :ρe)
+vars_state_for_transform(::AtmosModel) = (:ρ, :ρu, :ρv, :ρw)
+vars_transform(::AtmosModel) = (:u, :v, :w)
+vars_diffusive(::AtmosModel) = (:τ11, :τ22, :τ33, :τ12, :τ13, :τ23)
+
+function flux!(::AtmosModel, flux::Grad, state::State, diffstate::State, auxstate::State, t::Real)
+  # preflux
+  γ = γ_exact  
+  ρinv = 1 / state.ρ
+  u, v, w = ρinv * state.ρu, ρinv * state.ρv, ρinv * state.ρw
+  P = (γ-1)*(state.ρe - ρinv * (state.ρu^2 + state.ρv^2 + state.ρw^2) / 2)
+
+  # invisc terms
+  flux.ρ  = (state.ρu          , state.ρv          , state.ρw)
+  flux.ρu = (u * state.ρu  + P , v * state.ρu      , w * state.ρu)
+  flux.ρv = (u * state.ρv      , v * state.ρv + P  , w * state.ρv)
+  flux.ρw = (u * state.ρw      , v * state.ρw      , w * state.ρw + P)
+  flux.ρe = (u * (state.ρe + P), v * (state.ρe + P), w * (state.ρe + P))
+
+  # viscous terms
+  flux.ρu .-= (diffstate.τ11, diffstate.τ12, diffstate.τ13)
```

Comment:
With `\tau` as momentum flux tensor (see below), what should appear here is `-\rho\tau`.